src/Voron/GlobalFlushingBehavior.cs

```diff
@@ -242,7 +209,7 @@ private void FlushEnvironments(HashSet<StorageEnvironment> avoidDuplicates)
                     // we haven't reached the point where we have to flush, but we might want to, if we have enough 
                     // resources available, if we have more than half the flushing capacity, we can do it now, otherwise, we'll wait
                     // until it is actually required.
-                    if (_concurrentFlushes.CurrentCount > StorageEnvironment.MaxConcurrentFlushes / 2)
+                    if (_concurrentFlushes.CurrentCount < StorageEnvironment.MaxConcurrentFlushes / 2)
```

Comment:
Let's say that:

- CurrentCount: 2
- StorageEnvironment.MaxConcurrentFlushes: 20

2 < 10 - true, so we have enough resources available but we won't do the flush after this change

Reply:
Ouch. You're absolutely right. Very misleading name which made that we changed it quite recently CC @grisha-kotler

Reply:
Yes, I think we should rename this to be clearer